tribe-ext-calendar-widget-areas/index.php

```diff
@@ -0,0 +1,280 @@
+*/
+class Tribe__Extension__Calendar_Widget_Areas extends Tribe__Extension {
+	/**
+	 * Option key for which widget areas are enabled
+	 *
+	 * @var string
+	 */
+	protected $option_key_enabled_areas = 'tribe_ext_enabled_widget_areas';
+
+	/**
+	 * Setup the Extension's properties.
+	 *
+	 * This always executes even if the required plugins are not present.
+	 * 
+	 * @access: public
+	 */
+	public function construct() {
+		// Each plugin required by this extension
+		$this->add_required_plugin( 'Tribe__Events__Main' );
+				
+		// Set the extension's TEC URL
+		$this->set_url( 'https://theeventscalendar.com/extensions/calendar-widget-areas/' );
+	}
+	
+	/**
+	 * All available widget areas this extension supports and details about each in a multidimensional associative array.
+	 *
+	 * Follow this format to enable a new widget area for each action hook you want
+	 * to use. You will also need to add an additional method toward the end of this
+	 * file for each area added so we know what code to execute.
+	 * 
+	 * @access: public
+	 * @return: array
+	 */
+	public function get_all_areas() {
+		// format is:
+		// action_hook => array of details what to do if we use this hook
+		$areas = array(
+			'tribe_events_before_template'	=>	array(
+				'method'  	=> 'before_template',
+				'name'		=> __( 'TEC Above Calendar', 'tribe-extension' ),
+				'desc'		=> __( 'Widgets in this area will be shown ABOVE The Events Calendar.', 'tribe-extension' ),
+			),				
+			'tribe_events_after_template'	=>	array(
+				'method'  	=> 'after_template',
+				'name'		=> __( 'TEC Below Calendar', 'tribe-extension' ),
+				'desc'		=> __( 'Widgets in this area will be shown BELOW The Events Calendar.', 'tribe-extension' ),
+			),				
+			'tribe_events_before_view'	=>	array(
+				'method'  	=> 'before_view',
+				'name'		=> __( 'TEC Above Single Events', 'tribe-extension' ),
+				'desc'		=> __( 'Widgets in this area will be shown ABOVE Single Events.', 'tribe-extension' ),
+			),				
+			'tribe_events_after_view'	=>	array(
+				'method'  	=> 'after_view',
+				'name'		=> __( 'TEC Above Single Events', 'tribe-extension' ),
+				'desc'		=> __( 'Widgets in this area will be shown BELOW Single Events.', 'tribe-extension' ),
+			),
+		);
+		
+		return apply_filters( 'tribe_ext_calendar_widget_areas', $areas );
+	}
+	
+	/**
+	 * Build options to present to user
+	 * 
+	 * @access: public
+	 * @return: array
+	 */
+	public function get_available_area_options() {
+		$options = array();
+		
+		foreach ( $this->get_all_areas() as $key => $value ) {
+			$options[$key] = $value['name'];
+		}
+		
+		return $options;
+	}
+	
+	/**
+	 * The chosen widget areas to activate/run.
+	 * 
+	 * @access: public
+	 * @return: array
+	 */
+	public function get_enabled_areas_simple() {
+		$all_available = $this->get_all_areas();
+		
+		$enabled_areas = tribe_get_option( $this->option_key_enabled_areas, $all_available );
+		
+		if ( is_array( $enabled_areas ) ) {
```

Comment:
Wouldn't this always be an array? If so this is unnecessary.

Typically we avoid extra levels of nesting when possible. It makes code easier to read. Instead of doing `if( is_array() )` you could cast `$enabled_areas` to an array before it reached this block of code. `$enabled_areas = (array) tribe_get_option( ... )`

Then it will always be an array. Although I think it always will anyways, so this is probably unnecessary.